apps/system/test/unit/multi_screen_controller_test.js

```diff
@@ -312,4 +310,108 @@ suite('system/MultiScreenController', function() {
       });
     });
   });
+
+  suite('presentation device selection', function() {
+    var eventValidator;
+
+    function _content_event_handler(evt) {
+      if (eventValidator) {
+        eventValidator(evt);
+      }
+    }
+
+    setup(function() {
+      window.navigator.mozPresentationDeviceInfo = {
+        getAll: function() {
+          return Promise.resolve([
+              { name: 'test-name',
+                id: 'test-id',
+                type: 'test-type' }
+          ]);
+        }
+      };
+      window.addEventListener('mozContentEvent', _content_event_handler);
+    });
+
+    teardown(function() {
+      delete window.navigator.mozPresentationDeviceInfo;
+      window.removeEventListener('mozContentEvent', _content_event_handler);
+    });
+
+    test('device selected', function(done) {
+      eventValidator = function(evt) {
+        if (evt.detail.type !== 'presentation-select-result') {
+          return;
+        }
+
+        console.log(evt);
+        done(function() {
+          assert.equal(evt.detail.type, 'presentation-select-result');
+          assert.equal(evt.detail.deviceId, 'test-id');
+          assert.equal(evt.detail.id, 'test-selection-id');
+        });
+      };
+
+      this.sinon.stub(subject, 'showMenu', function() {
+        return Promise.resolve(0);
+      });
+
+      triggerMozChromeEvent({
+        type: 'presentation-select-device',
+        id: 'test-selection-id',
+      });
+    });
+
+    test('selection canceled', function(done) {
+      eventValidator = function(evt) {
+        if (evt.detail.type !== 'presentation-select-deny') {
+          return;
+        }
+
+        console.log(evt);
+        done(function() {
+          assert.equal(evt.detail.type, 'presentation-select-deny');
+          assert.equal(evt.detail.id, 'test-selection-id');
+        });
+      };
+
+      this.sinon.stub(subject, 'showMenu', function() {
+        return Promise.resolve(undefined);
+      });
+
+      triggerMozChromeEvent({
+        type: 'presentation-select-device',
+        id: 'test-selection-id',
+      });
+    });
+
+    test('canceled while no available device', function(done) {
+      window.navigator.mozPresentationDeviceInfo = {
+        getAll: function() {
+          return Promise.resolve([]);
+        }
+      };
+
+      eventValidator = function(evt) {
+        if (evt.detail.type !== 'presentation-select-deny') {
+          return;
+        }
+
+        console.log(evt);
+        done(function() {
+          assert.equal(evt.detail.type, 'presentation-select-deny');
+          assert.equal(evt.detail.id, 'test-selection-id');
+        });
+      };
+
+      this.sinon.stub(subject, 'showMenu', function() {
```

Comment:
It's unnecessary to stub "showMenu" in this case since it should be resolved with `undefined` automatically -- otherwise it's a bug and needs to be caught here.